cumulusci/tasks/salesforce/metadata_etl/base.py

```diff
@@ -0,0 +1,262 @@
+from cumulusci.utils import elementtree_parse_file, inject_namespace
+from cumulusci.core.config import TaskConfig
+
+
+class BaseMetadataETLTask(BaseSalesforceApiTask):
+    deploy = False
+    retrieve = False
+
+    namespaces = {"sf": "http://soap.sforce.com/2006/04/metadata"}
+
+    task_options = {
+        "unmanaged": {
+            "description": "If True, changes namespace_inject to replace tokens with a blank string"
+        },
+        "namespace_inject": {
+            "description": "If set, the namespace tokens in files and filenames are replaced with the namespace's prefix"
+        },
+        "api_version": {
+            "description": "Metadata API version to use, if not project__package__api_version."
+        },
+    }
+
+    def _init_options(self, kwargs):
+        super()._init_options(kwargs)
+
+        self.options["unmanaged"] = process_bool_arg(
+            self.options.get("unmanaged", False)
+        )
+        self.api_version = (
+            self.options.get("api_version")
+            or self.project_config.project__package__api_version
+        )
+
+    @property
+    def _namespace_injector(self):
+        return lambda text: inject_namespace(
+            "",
+            text,
+            self.options.get("namespace_inject"),
+            not self.options["unmanaged"],
+        )[1]
+
+    def _get_package_xml_content(self, deploy):
+        return f"""<?xml version="1.0" encoding="UTF-8"?>
+<Package xmlns="http://soap.sforce.com/2006/04/metadata">
+    <version>{self.api_version}</version>
+</Package>
+"""
+
+    def _generate_package_xml(self, deploy):
+        return self._namespace_injector(self._get_package_xml_content(deploy))
+
+    def _create_directories(self, tempdir):
+        if self.retrieve:
+            self.retrieve_dir = Path(tempdir, "retrieve")
+            self.retrieve_dir.mkdir()
+        if self.deploy:
+            self.deploy_dir = Path(tempdir, "deploy")
+            self.deploy_dir.mkdir()
+
+    def _retrieve(self):
+        api_retrieve = ApiRetrieveUnpackaged(
+            self, self._generate_package_xml(False), self.api_version
+        )
+        unpackaged = api_retrieve()
+        unpackaged.extractall(self.retrieve_dir)
+
+    def _transform(self):
+        pass
+
+    def _deploy(self):
+        target_profile_xml = Path(self.deploy_dir, "package.xml")
+        target_profile_xml.write_text(self._generate_package_xml(True))
+
+        api = Deploy(
+            self.project_config,
+            TaskConfig(
+                {
+                    "options": {
+                        "path": self.deploy_dir,
+                        "namespace_inject": self.options.get("namespace_inject"),
+                        "unmanaged": self.options.get("unmanaged"),
+                    }
+                }
+            ),
+            self.org_config,
+        )
+        result = api()
+        self._post_deploy(result)
```

Comment:
Should this be called from _run_task so that ETL tasks which override _deploy don't have to remember to call it?

Reply:
Yes. Fixing that.